pkg/controllers/management/clusterstats/statsaggregator.go

```diff
@@ -160,8 +165,31 @@ func (s *StatsAggregator) aggregate(cluster *v3.Cluster, clusterName string) err
 		v32.ClusterConditionNoMemoryPressure.False(cluster)
 	}
 
+	var oldVersion int
+	if cluster.Status.Version != nil {
+		oldVersion, err = strconv.Atoi(cluster.Status.Version.Minor)
+		if err != nil {
+			return err
+		}
+	}
 	versionChanged := s.updateVersion(cluster)
 
+	// If the cluster went through an upgrade from <=1.21 to >=1.22, restart
+	// the cluster agent in order to restart controllers that will no longer work
+	// with the new API.
+	if versionChanged {
+		newVersion, err := strconv.Atoi(cluster.Status.Version.Minor)
```

Comment:
Do we need to check `cluster.Status.Version != nil` here too?

Reply:
if `s.updateVersion` returns true then `cluster.Status.Version` is guaranteed to be set https://github.com/rancher/rancher/pull/35818/files/4e175ea23b1898d7924c5d52f171eb8dbc48fa37#diff-ae8f3a50a0e7a52898014aa299562a2cf77fb2ef40b5bc348478c6537a304978R213